### DISCUSSION TITLE
Missing required native library dependency: libsignal-client

### DISCUSSION THREAD

**Discussion by virtualed** [AsamK/signal-cli#840] [Q&A]
Hello,

I am stuck with this error msg trying to register signal on my computer without a smartphone.

- Win 7
- jdk-17_windows-x64
- signal-cli-0.10.0

Any help?
Thanx

---------------------------------

C:\Program Files\signal-cli-0.10.0\bin>signal-cli --verbose -u +3362825xxxx register

2021-12-18T18:54:28.910+0100 [main] WARN org.asamk.signal.manager.config.Service
Config - Failed to call libsignal-client: C:\Users\Administrateur\AppData\Local\Temp\resource3086590005915866371.so: Can't load this .dll (machine code=0x0) on a AMD 64-bit platform

Missing required native library dependency: libsignal-client

**Comment by AsamK** [comment]
See this wiki page: https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal

**Reply by virtualed** [reply]
Thanx for your reply

I copied the precompiled library (dll contained in signal_jni.dll-v0.11.0-x86_64-pc-windows.tar.gz) to both C:\Windows\System32 and C:\Windows\SysWOW64 (+ restart). But the error message is still the same.

The file "resource10039516872055313268.so" indeed does not exist in the specified path.

Any idea?
---------------------------------------

C:\Program Files\signal-cli-0.10.0\bin>signal-cli --verbose -u +3362825xxxx register 

2021-12-19T20:54:58.220+0100 [main] WARN org.asamk.signal.manager.config.ServiceConfig - Failed to call libsignal-client: C:\Users\Administrateur\AppData\Local\Temp\resource10039516872055313268.so: Can't load this .dll (machine code=0x0) on a AMD 64-bit platform

Missing required native library dependency: libsignal-client

**Reply by virtualed** [reply]
Thanx

I am stuck in step **2a** of the build with gradle as there is no such path "**build/install/signal-cli/bin**"
I have build/ classes or distributions or generated or libs or scripts of tmp

_1. Checkout the source somewhere on your filesystem with    
git clone https://github.com/AsamK/signal-cli.git

2. Execute Gradle:        
gradle build

2a. Create shell wrapper in *build/install/signal-cli/bin*:      
gradle installDist

2b. Create tar file in *build/distributions*:      
gradle distTar

2c. Create a fat tar file in *build/libs/signal-cli-fat*:
gradle fatJar

2d. Compile and run signal-cli:      
gradle run --args="--help"_

**Reply by exquo** [reply]
You can download signal-cli pre-built for windows:
https://github.com/exquo/signal-libs-build/releases/download/signal-cli_v0.10.0/signal-cli-v0.10.0-x86_64-Windows.tar.gz
